src/wallet/rpcwallet.cpp

```diff
@@ -1571,6 +1571,127 @@ UniValue listtransactions(const JSONRPCRequest& request)
+            "       {\n"
+            "         \"account\" : \"accountname\",      (string) DEPRECATED. The account name involved in the transaction, can be \"\" for the default account.\n"
+            "         \"address\" : \"address\",          (string) The dogecoin address involved in the transaction\n"
+            "         \"category\" : \"send|receive\",    (string) The category, either 'send' or 'receive'\n"
+            "         \"amount\" : x.xxx,                 (numeric) The amount in " + CURRENCY_UNIT + "\n"
+            "         \"label\" : \"label\",              (string) A comment for the address/transaction, if any\n"
+            "         \"vout\" : n,                       (numeric) the vout value\n"
+            "         \"fee\": x.xxx,                     (numeric) The amount of the fee in " + CURRENCY_UNIT + ". This is negative and only available for the \n"
+            "                                              'send' category of transactions.\n"
+            "         \"abandoned\": xxx                  (bool) 'true' if the transaction has been abandoned (inputs are respendable). Only available for the \n"
+            "                                              'send' category of transactions.\n"
+            "       }\n"
+            "       ,...\n"
+            "     ],\n"
+            "     \"hex\" : \"data\"         (string) Raw data for transaction\n"
+            "   }\n"
+            "]\n"
+
+            "\nExamples:\n"
+            "\nList all transactions hashes in the wallet that do not have a mempool entry.\n"
+            + HelpExampleCli("liststucktransactions", "") +
+            "\nList all transactions hexs in the wallet that do not have a mempool entry \n"
+            + HelpExampleCli("liststucktransactions", "1") +
+            "\nAs a json rpc call\n"
+            + HelpExampleRpc("liststucktransactions", "")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    int verbosity = 0;
+    if (request.params.size() > 0)
+        verbosity = request.params[0].get_int();
+
+    isminefilter filter = ISMINE_SPENDABLE;
+    if(request.params.size() > 1)
+        if(request.params[1].get_bool())
+            filter = filter | ISMINE_WATCH_ONLY;
+
+    if(verbosity < 0 || verbosity > 2)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid verbosity");
+
+    UniValue ret(UniValue::VARR);
+
+    const CWallet::TxItems & txOrdered = pwalletMain->wtxOrdered;
+
+    for (CWallet::TxItems::const_iterator it = txOrdered.begin(); it != txOrdered.end(); ++it)
+    {
+        CWalletTx *const wtx = (*it).second.first;
+        if (wtx->GetDepthInMainChain() <= 0 && !wtx->InMempool())
+        {
+            if (verbosity == 0) {
+                ret.push_back(wtx->GetHash().ToString());
+            } else if (verbosity == 1) {
+                string strHex = EncodeHexTx(static_cast<CTransaction>(*wtx), RPCSerializationFlags());
+                ret.push_back(strHex);
```

Comment:
Currently these verbosity formats include `watchonly` even if `include_watchonly` is `false` (explicitly or implicitly.)

Perhaps we can add the `IsFromMe()` condition at the top level `if` statement on line 1660:

```cpp
if (wtx->GetDepthInMainChain() <= 0 && !wtx->InMempool() && wtx->IsFromMe(filter)) {
...
```

and can then remove the other evaluations of `wtx->IsFromMe(filter)`, as that will always be true.

This means that we only show our own, outgoing transactions, but since these are the only ones we are capable of doing something about (`abandontransaction` and ideally re-spend the inputs) as for incoming coin we can in nearly any scenario only rebroadcast and hope for the best (fee and dust limits override CPFP), so this would cover the need discussed in #2721 (being aware of any transactions that may go out when upgrading / lowering dust limits) for a first implementation.

We may be able to expand this functionality a bit in a future iteration, to get more fine-grained controls. The must-have for this feature is knowing which of our transactions are stuck (that we created with 1.14.0), so that we can do something about them.

Reply:
I completely overlooked that. The changes sound good to me. :+1: I believe it's important to have the core feature working as discussed in #2721.